### PR TITLE
Misc. Update

### PR DESCRIPTION
**Pull request by Expertcoderz**
I did the following:

### Revamped server details UI
Replaced the simple list with a tabbed UI window showing even more info about the game.

### Hide chat commands
Added a new client setting that allows players to auto-hide commands run from the chat (without using /e), like in PB Kronos.

### (Hopefully) fixed !addfriend and !unfriend
Gotta keep testing.

### Very minor stuff
Fixed an insignificant typo in the player inspection UI module and improved the name display format of the online friends UI.

### CODE REVIEW

File: MainModule/Client/Core/Variables.lua [hunk 1]
```diff
@@ -80,4 +81,4 @@ return function()
 			FogStart = service.Lighting.FogStart;
 		}
 	};
-end
\ No newline at end of file
+end
```
Comment (sasial-dev) [inline MainModule/Client/Core/Variables.lua:84]: Why did you remove the new line at the end of the file?
It's a standard programming convention.
```suggestion
end

```

File: MainModule/Client/UI/Default/Settings.lua [hunk 2]
```diff
@@ -65,6 +65,21 @@ return function(data)
 				toggle.Text = text
 			end
 		};
+		{
+			Text = "Hide Chat Commands: ";
+			Desc = "- Hide your commands run from the chat from others";
```
Comment (sasial-dev) [inline MainModule/Client/UI/Default/Settings.lua:70]: Maybe reword for clarity?
```suggestion
			Desc = "- Hide commands ran from the chat";
```

File: MainModule/Client/UI/Default/Settings.lua [hunk 3]
```diff
@@ -202,4 +217,4 @@ return function(data)
 		gTable = window.gTable
 		window:Ready()
 	end
-end
\ No newline at end of file
+end
```
Comment (sasial-dev) [inline MainModule/Client/UI/Default/Settings.lua:220]: Why did you remove the new line at the end of the file?
It's a standard programming convention.
```suggestion
end

```

File: MainModule/Client/UI/Default/UserPanel.lua [hunk 4]
```diff
@@ -1127,6 +1127,21 @@ return function(data)
 						toggle.Text = text
 					end
 				};
+				{
+					Text = "Hide Chat Commands: ";
+					Desc = "- Hide your commands run from the chat from others";
```
Comment (sasial-dev) [inline MainModule/Client/UI/Default/UserPanel.lua:1132]: Maybe reword for clarity?
```suggestion
					Desc = "- Hide commands ran from the chat";
```

File: MainModule/Server/Core/Process.lua [hunk 5]
```diff
@@ -384,14 +391,20 @@ return function(Vargs)
...
-
 					if Settings.ChatCommands then
-						if msg:sub(1,3)=="/e " then
+						if Remote.Get(p,"Function","ChatCmdsHidden") == true and (msg:sub(1,1) == Settings.Prefix or msg:sub(1,1) == Settings.PlayerPrefix) then
+							Remote.Send(p,"Function","ChatMessage","> "..msg,Color3.new(255, 255, 255))
+							Process.Command(p,msg,{Chat = true;})
+						elseif msg:sub(1,3)=="/e " then
+							service.Events.PlayerChatted:Fire(p,msg)
 							msg = msg:sub(4)
+							Process.Command(p,msg,{Chat = true;})
```
Comment (sasial-dev) [inline MainModule/Server/Core/Process.lua:401]: I know this isn't part of your PR, but it might be worth allowing support for "/system "
```suggestion
							Process.Command(p,msg,{Chat = true;})
						elseif msg:sub(1,8)=="/system " then
							service.Events.PlayerChatted:Fire(p,msg)
							msg = msg:sub(9)
							Process.Command(p,msg,{Chat = true;})
```

**Review by sasial-dev — CHANGES_REQUESTED** [review]
Nice Job!
I was trying to look for code changes/feedback but only managed to come up with some style/text changes.

File: MainModule/Client/UI/Default/Settings.lua (hunk 3 above)
Comment (ccuser44) [reply MainModule/Client/UI/Default/Settings.lua:220]: I think github does this automaticly

File: MainModule/Client/UI/Default/UserPanel.lua (hunk 4 above)
Comment (happyman090) [reply MainModule/Client/UI/Default/UserPanel.lua:1132]: "Hide your chat commands from others" sounds better than both of them in my opinion

"Hide commands ran from the chat" implies that commands sent by others will also be hidden when that isn't the case
"Hide your commands run from the chat from others" is intricate but not necessarily unclear in my opinion

File: MainModule/Client/UI/Default/Settings.lua (hunk 3 above)
Comment (Expertcoderz) [reply MainModule/Client/UI/Default/Settings.lua:220]: I don't remember doing that. It's either an accident or GitHub behavior.

**Comment by Expertcoderz** [comment]
@sasial-dev @happyman090 @ccuser44
Thanks for your suggestions/comments. I will publish the appropriate improvements soon.
